### ISSUE TITLE
DBreeze.Exceptions.DBreezeException: Creation of the database folder failed! ---> DBreeze.Exceptions.DBreezeException: Database is not operable, please find out the problem and restart the engine! FSR INIT FAILED

### ISSUE THREAD

**Issue by AbdisamadMoh** [hhblaze/DBreeze#59]
```
DBreeze.Exceptions.DBreezeException: Creation of the database folder failed! ---> DBreeze.Exceptions.DBreezeException: Database is not operable, please find out the problem and restart the engine! FSR INIT FAILED: C:\Users\user\Documents\Visual Studio 2015\Projects\chat\chat\App_Data/D12345/Message/MessageID/Abdi\_DBreezeSchema ---> System.IO.IOException: The process cannot access the file 'C:\Users\user\Documents\Visual Studio 2015\Projects\chatchat\App_Data\D12345\Message\MessageID\Abdi\_DBreezeSchema' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at DBreeze.Storage.FSR.InitFiles()
   --- End of inner exception stack trace ---
   at DBreeze.Storage.FSR.InitFiles()
   at DBreeze.Storage.FSR..ctor(String fileName, TrieSettings trieSettings, DBreezeConfiguration configuration)
   at DBreeze.Storage.StorageLayer..ctor(String fileName, TrieSettings trieSettings, DBreezeConfiguration configuration)
   at DBreeze.Scheme.OpenSchema()
   at DBreeze.Scheme..ctor(DBreezeEngine DBreezeEngine)
   at DBreeze.DBreezeEngine.InitDb()
   --- End of inner exception stack trace ---
   at DBreeze.DBreezeEngine.InitDb()
   at DBreeze.DBreezeEngine..ctor(String DBreezeDataFolderName)
   at Chat.Main.Main.SendMessage(String _Id, String _msg, String _from, String _to) in C:\Users\user\Documents\Visual Studio 2015\Projects\chat\chat\Main\Main.cs:line 60
```

Am getting that error and its seems with file inaccessibility. It happens when i try to concurrently write or read database file from different Threads(From different Instant of **DBreezeEngine**). Im using ASP.NET Webforms and every request will have different Instant of **DBreezeEngine**.

I tried to use **DBIsoperable** but the error happens before it.

Is there away to concurrently read/write db file from different **DBreezeEngine** instant?

**Comment by hhblaze** [comment]
Clear, the problem is that DBEngine should be a static variable and must be instantiated only once somewhere in static file. Then you can use this static engine on each page while its rendering. Rule: engine must be instantiated only once and used within the whole program life cycle from different threads.

**Comment by Hadaa12** [comment]
That worked well. Thank you very much

**Comment by shapuday** [comment]
how do i put DBEngine to static variable?

**Comment by hhblaze** [comment]
> how do i put DBEngine to static variable?

Initialize once and access this DBEngine type variable from the whole software - that is all you need.

**Comment by shapuday** [comment]
> > how do i put DBEngine to static variable?
> 
> Initialize once and access this DBEngine type variable from the whole software - that is all you need.

idunno how to do that i just want to work my ds3 controller on my pc

**Comment by RoarkDude** [comment]
No, I have to be able to change the Data Folder during the lifespan of the app.  This used to work but it broke at some point.

```

            Static db As DBreeze.DBreezeEngine = Nothing
            db = New DBreeze.DBreezeEngine("C:\Users\Owner\AppData\Local\QIntegrator\eCricket")
            db.Dispose()
            db = New DBreeze.DBreezeEngine("C:\Users\Owner\AppData\Local\QIntegrator\eCricket")
            db.Dispose()
```

**Comment by hhblaze** [comment]
Fixed in DBreeze 1.118.2024.1016 (get it via NuGet or whatever)

**Comment by RoarkDude** [comment]
Amazing! Thank you!
